crates/anvil-polkadot/src/substrate_node/service/mod.rs

```diff
@@ -30,7 +31,7 @@ mod consensus;
 mod executor;
 pub mod storage;
 
-pub type Backend = sc_service::TFullBackend<Block>;
+pub type Backend = LazyLoadingBackend<Block>;
```

Comment:
the caveat of using the lazy loading backend even if forking is not configured is that we'll end up storing everything in memory,
instead if using the substrate DB.
This is probably fine, since it's a test node after all. Moreover, anvil also stores everything in memory unless configured to prune.

Reply:
Any final decision on this? 🤔

Reply:
is there an easy way in which we could make the Backend a `dyn` trait object which under the hood would be able to use the regular backend or the lazy-loading one depending on how it's configured via CLI?

Reply:
I created this issue to investigate about the multiple backend impl support https://github.com/ChainSafe/gossamer-parity/issues/80